### ISSUE TITLE
CoreML .mlpackage model exported with task="detect" is misinterpreted as obb in Flutter (ultralytics_yolo)

### ISSUE THREAD

**Issue by keerthivasan-thiran** [ultralytics/ultralytics#21571]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

# ❗️ Flutter iOS Detection Task Misinterpreted as OBB in `.mlpackage`

## Summary

After training a YOLOv8n model using the Ultralytics Python API and exporting it to CoreML format with `task="detect"`, the resulting `.mlpackage` is **misinterpreted as an `obb` (oriented bounding box)** model in Flutter when using the `ultralytics_yolo` plugin.

The same model, when exported to `.tflite`, works correctly as a `detect` model on Android.

---

## ✅ Working Setup (Android)

```
from ultralytics import YOLO

model = YOLO("yolov8n.pt")
model.train(
    data='/content/PPE-1/data.yaml',
    epochs=50,
    imgsz=640,
    batch=64,
    optimizer='SGD',
    patience=10,
    lr0=0.01
)

# Export to TFLite for Android
model.export(format="tflite", task="detect")
```

✅ The `.tflite` model behaves as expected on Android.

---

## ❌ Non-working Setup (iOS - CoreML)

```
from ultralytics import YOLO

# Load best model checkpoint
model = YOLO("/content/runs/detect/train/weights/best.pt")

# Export to CoreML format for iOS
model.export(format="coreml", task="detect")
```

🔴 The `.mlpackage` model loads in Flutter but is **interpreted as an `obb` model**, not `detect`.

## 💻 Environment

- `ultralytics_yolo` pub version: 0.1.18
- Flutter version : 3.27.0
- Platform: iOS  
- Model: `yolov8n.pt`, trained on custom dataset

---

Hope any suggestion to solve the issue is appreciated.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @keerthivasan-thiran, thank you for your detailed report and for using Ultralytics 🚀! We appreciate you outlining your process and observations.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) if you haven’t already. This helps our team reproduce and debug the issue more efficiently. If you can share a small sample project, model, or relevant logs, that would be very helpful.

In the meantime, we recommend checking the [Docs](https://docs.ultralytics.com/) for export and deployment guidance. For more on YOLO model tasks and export options, see [Python Usage](https://docs.ultralytics.com/usage/python/) and [CLI Usage](https://docs.ultralytics.com/usage/cli/).

You’re also welcome to join our community for real-time discussions or support:
- [Discord](https://discord.com/invite/ultralytics) 🎧
- [Discourse](https://community.ultralytics.com/)
- [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

First, please ensure you’re using the latest `ultralytics` package (and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in these up-to-date verified environments:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across platforms every 24 hours and on every commit.

---

This is an automated response 🤖. An Ultralytics engineer will review your issue and assist you soon. Thank you for your patience and for helping us improve Ultralytics HUB and YOLO!

**Comment by Y-T-G** [comment]
What does it mean that it's interpreted as OBB?

**Comment by keerthivasan-thiran** [comment]
@Y-T-G , Which is when the gave the task : YOLOTask.obb in Flutter code, I'm seeing some Boxes in the UI, if the give the task: YOLOTask.detect, no results in iOS

**Comment by Y-T-G** [comment]
Did you try using `nms=True` during export?

**Comment by keerthivasan-thiran** [comment]
@Y-T-G  This solution works for me. I tried `nms=True`, then I can able to get the detection in my iOS app. 

Final Working Command : `model.export('coreml',task='detect',nms=True)`

Thank you @Y-T-G
